src/components/document/student-group-view.tsx

```diff
@@ -43,13 +43,25 @@ export const StudentGroupView:React.FC<IProps> = ({ groupId, setGroupId }) => {
   const [focusedGroupUser, setFocusedGroupUser] = useState<GroupUserModelType | undefined>();
   const selectedGroupId = groupId || (groups.allGroups.length ? groups.allGroups[0].id : "");
   const groupUsers = getGroupUsers(user.id, groups, documents, selectedGroupId);
-  const group = groups.getGroupById(selectedGroupId);
-
   const ui = useUIStore();
+  const group = groups.getGroupById(selectedGroupId);
   const isStudentViewActiveTab = (ui.activeNavTab === "student-work");
   const isChatPanelShown = ui.showChatPanel;
   const shrinkStudentView  = isStudentViewActiveTab && isChatPanelShown;
-  const classes = classNames("document", "student-group-view", {"shrink-student-view": shrinkStudentView});
+  const documentSelectedForComment = ui.showChatPanel && ui.selectedTileIds.length === 0 && ui.focusDocument;
+  const studentGroupViewClasses = classNames( "editable-document-content", "document", "student-group-view",
+  {"shrink-student-view": shrinkStudentView}, {"comment-select" : documentSelectedForComment});
+
+  if (focusedGroupUser && isStudentViewActiveTab){
+    const id = focusedGroupUser.id;
+    const foundUser = groupUsers.find(obj => obj.user.id === id);
+    if (foundUser && foundUser.doc && foundUser.doc.key && foundUser.doc.groupId){
+      //use group buttons (G1, G2, etc) as a pseudo subTab
+      const groupSubTab = foundUser.doc.groupId;
+      const documentKey = foundUser.doc.key;
+      ui.openSubTabDocument("student-work", groupSubTab, documentKey);
```

Comment:
It is a odd that a component is opening a sub tab document in its render function. Updating the UI store like this should usually be done in an event handler. And then the component would be using the state in the UI store to decide what to render.